### ISSUE TITLE
Python 3 compatibility

### ISSUE THREAD

**Issue by herlo** [CentOS-PaaS-SIG/linchpin#283]
Rework anything that doesn't meet the Python 3.6 requirements to be compatible. The code needs to continue to be 2.7+ compatible as well.

See https://docs.python.org/3/howto/pyporting.html

**Comment by herlo** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Python 3 conversion](https://trello.com/c/yldI3FMK/18-python-3-conversion)

**Comment by jpryor-rh** [comment]
Looks to me that beaker-client is not Python 3 compatible as I found out a few days ago on Fedora 26 with Python3:
```
(linchpin) [root@fed26 work]# pip install linchpin[beaker]
<snipped>
      File "/tmp/pip-build-e57v6sgz/beaker-client/setup.py", line 3, in <module>
        import commands
    ModuleNotFoundError: No module named 'commands'
       ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-e57v6sgz/beaker-client/
```

Beaker devs are aware of it and it is tracked in bugzilla at https://bugzilla.redhat.com/show_bug.cgi?id=1455424

**Comment by herlo** [comment]
Correct. Beaker is not Python3 ready. The conversation has been brought up multiple times. One of the considerations to combat this is to have the rest of LinchPin be Python3 ready, and then have provider modules (like beaker) as add-on components. This is more of a long term plan, however.

**Comment by samvarankashyap** [comment]
Follow up with beaker team.

**Comment by samvarankashyap** [comment]
Create a community module to the beaker and push it upstream.

**Comment by mrniranjan** [comment]
@samvarankashyap  any update on this. If required i can test python3 compatibility

**Comment by samvarankashyap** [comment]
@samvarankashyap Add compatibility to both nightly and integration tests. create issues around it.

**Comment by samvarankashyap** [comment]
Beaker 27 is released we are unblocked on python3. 
Waiting for beaker 27 to be pushed to PyPI. 
Need to create a ticket for hosting carbon for running downstream and upstream nightly tests.

**Comment by samvarankashyap** [comment]
Update: We are unblocked on beaker 27 release since it is currently available on PyPi.

**Comment by samvarankashyap** [comment]
Beaker 27 is currently a dependency for linchpin. 
Linchpin is currently python3 compatible since 1.9.1.1 release. 
I believe this ticket can be closed in perspective of python3 compatibility. 
We will be creating another ticket to move all tests to all versions python3. 
Currently:
unit tests run on python3
integration tests run on python3 on feodra31.
